### PR TITLE
Repair quotient rings

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
CC: @wdecker , @fingolfin @fieker

### CODE REVIEW

**Comment by thofma** [comment]
Was it decided at one of the Kaiserslautern meetings, that we'd rather have a hash functions which errors than a slow default one?

**Comment by HechtiDerLachs** [comment]
Yes. This was @fieker 's proposal.

**Comment by lgoettgens** [comment]
> Was it decided at one of the Kaiserslautern meetings, that we'd rather have a hash functions which errors than a slow default one?

This is similar to https://github.com/oscar-system/GAP.jl/pull/891.

**Comment by HechtiDerLachs** [comment]
Looks like the code coverage report is nothing that can be taken serious these days. All coverage changes occur in places that I did not touch; not even indirectly! I suppose this is due to the various failing tests? 

Either way: I would like to point to #2789 if anyone feels like doing a proper review.

File: src/Rings/MPolyQuo.jl [hunk 1]
```diff
@@ -807,9 +842,20 @@ true
...
-  return f.f == g.f
+  Q = parent(f)::MPolyQuoRing
+  P = base_ring(Q)::MPolyRing
+  kk = coefficient_ring(P)::Ring
+
+  # The following bracket is for the default assumption
+  if is_coefficient_type_with_groebner_basis_algorithm_in_singular(kk)
+    f.f == g.f && return true
+    hash(f) == hash(g) && return true # calls simplify already
+    return f.f == g.f
```
Comment (simonbrandhorst) [inline src/Rings/MPolyQuo.jl:853]: This looks wrong to me. Please double check

Comment (HechtiDerLachs) [reply src/Rings/MPolyQuo.jl:853]: Indeed. What was I thinking...? Corrected in #2789 .

**Pull request closed without merging**